### PR TITLE
Search by office fix

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
#### Which User story does this relate to (link)?
-N/A

#### Brief feature description
- Search by email / firstName / racfid is blocking due to the query passed in handleOfficeChange method. 
- Input field throws a warning `Input is changing uncontrolled input of type text to be controlled` since the value of fields are undefined in initial state.
- Search by case-insensitive racfid

### CODE REVIEW

File: app/javascript/containers/searchUserListContainer.js [hunk 1]
```diff
@@ -41,10 +41,10 @@ function mapStateToProps(state) {
     error: searchUserList.error,
     officesList: officesList(state),
     inputData: searchUserList.inputData,
-    firstName: searchUserList.inputData.firstName,
-    email: searchUserList.inputData.email,
-    CWSLogin: searchUserList.inputData.CWSLogin,
-    lastName: searchUserList.inputData.lastName,
+    firstName: searchUserList.inputData.firstName ? searchUserList.inputData.firstName : '',
```
Comment (kelfink) [inline app/javascript/containers/searchUserListContainer.js]: doesn't safeGet do this for us?  import safeGet from 'lodash.get' then...
```suggestion
   firstName: safeGet(searchUserList, 'inputData.firstName', '')
```

Comment (Sushmareddy-OSI) [reply app/javascript/containers/searchUserListContainer.js]: yeah! good call out, isn't better to use the safeGet in the selector instead here in container?

Comment (kelfink) [reply app/javascript/containers/searchUserListContainer.js]: Sure.  That sounds even better.

**Comment by Sushmareddy-OSI** [comment]
retest this please

File: lib/elastic/user_query_builder.rb [hunk 2]
```diff
@@ -54,7 +54,8 @@ def self.query_leaves(query)
         { conjunction: 'AND', query: { match: { 'email.keyword': value } } } unless value.empty?
       end,
       racfid: lambda do |value|
-        { conjunction: 'AND', query: { match: { 'racfid.keyword': value } } } unless value.empty?
+        { conjunction: 'AND', query: { match: { 'racfid.keyword': value.upcase } } } unless value
```
Comment (kelfink) [inline lib/elastic/user_query_builder.rb]: do we need the same logic on email.keyword, to upcase it ?

Comment (Sushmareddy-OSI) [reply lib/elastic/user_query_builder.rb]: No not for the email.

**Pull request merged**